build-info-api/src/main/java/org/jfrog/build/api/builder/BuildBuilder.java

```diff
@@ -0,0 +1,403 @@
+        return this;
+    }
+
+    /**
+     * Sets the URL of the build
+     *
+     * @param url Build URL
+     * @return Builder instance
+     */
+    public BuildBuilder url(String url) {
+        this.url = url;
+        return this;
+    }
+
+
+    /**
+     * Sets the parent build name of the build
+     *
+     * @param parentName Build parent build name
+     * @return Builder instance
+     */
+    public BuildBuilder parentName(String parentName) {
+        this.parentName = parentName;
+        return this;
+    }
+
+    /**
+     * Sets the parent build number of the build
+     *
+     * @param parentNumber Build parent build number
+     * @return Builder instance
+     */
+    public BuildBuilder parentNumber(String parentNumber) {
+        this.parentNumber = parentNumber;
+        return this;
+    }
+
+    /**
+     * Sets the vcs revision (format is vcs specific)
+     *
+     * @param vcs The vcs data
+     * @return Builder instance
+     */
+    public BuildBuilder vcs(List<Vcs> vcs) {
+        this.vcs = vcs;
+        return this;
+    }
+
+    /**
+     * Sets the vcs revision (format is vcs specific)
+     *
+     * @param vcsRevision The vcs revision
+     * @return Builder instance
+     */
+    public BuildBuilder vcsRevision(String vcsRevision) {
+        this.vcsRevision = vcsRevision;
+        return this;
+    }
+
+    /**
+     * Sets the vcs revision (format is vcs specific)
+     *
+     * @param vcsUrl The vcs revision
+     * @return Builder instance
+     */
+    public BuildBuilder vcsUrl(String vcsUrl) {
+        this.vcsUrl = vcsUrl;
+        return this;
+    }
+
+    /**
+     * Sets the modules of the build
+     *
+     * @param modules Build modules
+     * @return Builder instance
+     */
+    public BuildBuilder modules(ConcurrentHashMap<String, Module> modules) {
+        this.modules = modules;
+        return this;
+    }
+
+    /**
+     * Sets the modules of the build
+     *
+     * @param modules Build modules
+     * @return Builder instance
+     */
+    public BuildBuilder modules(List<Module> modules) {
+        ConcurrentHashMap<String, Module> modulesMap = new ConcurrentHashMap<String, Module>();
+        for (Module module : modules) {
+            modulesMap.put(module.getId(), module);
+        }
+
+        this.modules = modulesMap;
+        return this;
+    }
+
+    public BuildBuilder statuses(List<PromotionStatus> statuses) {
+        this.statuses = statuses;
+        return this;
+    }
+
+    public BuildBuilder addStatus(PromotionStatus promotionStatus) {
+        if (statuses == null) {
+            statuses = new ArrayList<>();
+        }
+        statuses.add(promotionStatus);
+        return this;
+    }
+
+    /**
+     * Sets the violation notifications of the build
+     *
+     * @param licenseControl Build violation  recipients.
+     * @return Builder instance
+     */
+    public BuildBuilder licenseControl(LicenseControl licenseControl) {
```

Comment:
This method, in addition to everything that is around license control should be removed. I suggest we delete the LicenseControl class, and then remove everything that uses it.